### ISSUE TITLE
Add prettier formatting rules and enable auto-format

### ISSUE THREAD

**Issue by solomondefi** [solomondefi/solomon-monorepo#7]
Add prettier rules based on previous projects, and auto-format on save. VSCode extension recommendation is already in place.

### Tasks
- Review prettier & eslint rules and compare to our existing projects
- Add Webstorm setup instructions (format on save, etc)
- Add Git commit hooks

**Comment by apolkingg8** [comment]
Please provide the rules based on previous projects and assign to me. I'll do it after #8 done.

**Comment by solomondefi** [comment]
Sure, here are some guidelines:

Add `.prettierignore`
```
# Add files here to ignore them from prettier formatting

/dist
**/artifacts
**/coverage
node_modules
*-lock.*
cache
logs
log
```

Add `.prettierrc`
```
{
  "singleQuote": true,
  "semi": false,
  "trailingComma": "all",
  "arrowParens": "always",
  "printWidth": 90
}
```

Update `.vscode/settings.json` to include the following rules. Since you use Webstorm feel free to add similar rules for that if needed.
```
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true
```

It may be necessary to extend `prettier` in eslint config to avoid conflicts. Not sure if Nx eslint base already takes care of this.

**Comment by solomondefi** [comment]
Closed in #11
